lib/ts/recipe/thirdpartyemailpassword/components/themes/signInAndUp/index.tsx

```diff
@@ -45,19 +51,23 @@ const SignInAndUpTheme: React.FC<ThirdPartyEmailPasswordSignInAndUpThemeProps> =
                     setIsSignUp={(isSignUp) => props.epDispatch({ type: isSignUp ? "setSignUp" : "setSignIn" })}
                 />
                 {props.commonState.error && <GeneralError error={props.commonState.error} />}
-                {props.tpChildProps !== undefined && (
+                {props.tpChildProps !== undefined && thirdPartyEnabled && (
                     <ProvidersForm {...props.tpChildProps} featureState={props.tpState} dispatch={props.tpDispatch} />
                 )}
-                {props.config.disableEmailPassword !== true && props.thirdPartyRecipe !== undefined && (
-                    <div data-supertokens="thirdPartyEmailPasswordDivider">
-                        <div data-supertokens="divider"></div>
-                        <div data-supertokens="thirdPartyEmailPasswordDividerOr">
-                            {t("THIRD_PARTY_EMAIL_PASSWORD_SIGN_IN_AND_UP_DIVIDER_OR")}
+                {props.config.disableEmailPassword !== true &&
+                    emailPasswordEnabled &&
+                    thirdPartyEnabled &&
+                    props.thirdPartyRecipe !== undefined && (
```

Comment:
since we have `emailPasswordEnabled` and `thirdPartyEnabled` these (and the following, i.e.: lines 54 & 70) checks should be consolidated into those booleans.